### PR TITLE
add demo for go-plugin codec

### PR DESCRIPTION
**Pull request by fdingiit**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: examples/codes/mini-thrift-with-xprotocol-goplugin-sample/thrift/codec/api.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+package main
+
+import (
+	"mosn.io/api"
+	"mosn.io/pkg/buffer"
+)
+
+// NewRpcRequest is a utility function which build rpc Request object of thrift protocol.
+func NewRpcRequest(requestId uint32, headers api.HeaderMap, data buffer.IoBuffer) *Request {
```
Comment (nejisama) [inline examples/codes/mini-thrift-with-xprotocol-goplugin-sample/thrift/codec/api.go:26]: 直接用api.IoBuffer吧，没有历史包袱就不用buffer.IoBuffer这种兼容模式的了，相当于也是推荐后面的人直接用api

File: examples/codes/mini-thrift-with-xprotocol-goplugin-sample/thrift/codec/buffer.go [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+
+	*buf = thriftBuffer{}
+}
+
+type thriftBuffer struct {
+	request  Request
+	response Response
+}
+
+func bufferByContext(ctx context.Context) *thriftBuffer {
```
Comment (nejisama) [inline examples/codes/mini-thrift-with-xprotocol-goplugin-sample/thrift/codec/buffer.go:352]: 这个内存复用不是必须的，可以直接make Request / Response 对象就可以，example没有必要搞那么复杂

File: examples/codes/mini-thrift-with-xprotocol-goplugin-sample/thrift/codec/command.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+	rawContent []byte // sub slice of raw data, content bytes
+
+	Data    buffer.IoBuffer // wrapper of raw data
+	Content buffer.IoBuffer // wrapper of raw content
+
+	ContentChanged bool // indicate that content changed
+}
+
+func (r *Response) GetTimeout() int32 {
+	return 10
```
Comment (nejisama) [inline examples/codes/mini-thrift-with-xprotocol-goplugin-sample/thrift/codec/command.go:159]: 这个单位是ms，默认返回10 感觉不合适，如果不知道返回什么就返回0比较好

**Review by nejisama — APPROVED** [review]

**Comment by taoyuanyuan** [comment]
@nejisama  repalced by https://github.com/mosn/mosn/pull/1622
